extra_tests/test_startup.py

```diff
@@ -7,3 +7,23 @@ def test_platform_not_imported():
     modules = [x.strip(' "\'') for x in out.strip().strip('[]').split(',')]
     assert 'platform' not in modules
     assert 'threading' not in modules
+
+
+def test_executable_win32():
+    # issue #4003
+    import os
+    import subprocess
+    import sys
+
+    out = subprocess.check_output([r'*', '-c',
+         'import sys; print (repr(sys.executable))'], universal_newlines=True, executable=sys.executable)
+    if sys.platform == 'win32':
+        assert out.strip() == repr(sys.executable)
+
+        fake_executable = r'C:\Windows\System32\cmd.exe'
+        assert os.path.isfile(fake_executable)  # should exist always
+        out = subprocess.check_output([fake_executable, '-c',
+            'import sys; print (repr(sys.executable))'], universal_newlines=True, executable=sys.executable)
+        assert out.strip() == repr(sys.executable)
```

Comment:
Checking `extra_tests/test_startup.py` with CPython, this will need to be changed for py3.11.